content/kips/kip-18.md

```diff
@@ -0,0 +1,69 @@
+
+## Abstract
+
+An abstraction layer to be built on top of the Synthetix Futures (V1) protocol. It enables the creation of "margin accounts" for each trader. Margin accounts enable Kwenta to provide cross margin functionality.
+
+## Motivation
+
+The current Futures system requires traders to deposit margin into specific markets before trading. This two step process causes friction when trying to act on market conditions and when managing margin between multiple markets. Cross margin improves the UX of trading on Kwenta.
+
+## Specification
+
+A major change introduced with this KIP is the introduction of a "margin account" system. Each trader will be required to set up a smart contract wallet (this step is managed by the UI) that holds margin on behalf of the trader. Smart contract accounts also open the door for future functionality as seen by [KIP-19: Advanced Orders](./kip-19.md).
+
+Cross margin is achieved by bundling together the margin step and the trade step when modifying a futures position. With the current implementation a trader can also open/modify/close multiple positions at once (atomically). It will be up to the frontend implementation on how to manage and rebalance margin. 
+
+### Rationale & User Flows
+
+- As a **trader** I want:
+    - To gain exposure to a **singular asset**:
+        - By opening BTC position without having to perform two transactions.
+        - By opening BTC position using the margin from closing a LINK position.
+    - To gain exposure to **multiple assets**:
+        - By opening BTC and ETH position without having to perform multiple transactions.
+    - To gain exposure on the **negative correlation of two assets**:
+        - By opening pair trade ETH/SOL in one step.
+        - By rebalancing a pair trade with profits from a drifting asset.
+
+### Technical Specification
+
+There will be a factory contract (MarginAccountFactory) that creates new margin accounts. Each margin account's functionality will be restricted to the owner of the account, which is the original creator.
+
+The current architecture relies on MinimalProxies where each margin account is a minimal proxy. Events are emitted on account creation.
+
+`distributeMargin` is the core function of Cross Margin and accepts an array of `UpdateMarketPositionSpec` structs. 
+
+```
+function distributeMargin(UpdateMarketPositionSpec[] memory _newPositions);
+``` 
+
+Each spec will specify the market, margin Δ, size Δ, and optional* boolean to denote positions being closed. 
+
+A positive margin Δ will deposit margin into a specified market and a negative margin delta will withdraw margin from a specified market. 
+
+A positive size Δ will increase the notional size of a position and negative size Δ will decrease the notional size or invert the direction of a position (long vs. short).
+
+*Note that positions can still be closed if the passed size Δ is the exact inverse of the current size. 
+
```

Comment:
Another phase we discussed is making the `EOA -> Smart contract account` mapping one-to-many.

Currently there would be a restriction on opening multiple pair trades using the same asset, for example. A user is limited to opening a single ETH long or short until they can make many smart contract accounts.

Reply:
> Might be good to add a section around UI / UX with some of the points we discussed previously:

Decided not to address these here since they pertain more towards internal plan for implementation. And not 100% sure if we're going to cover UX processes in KIPs. cc @Andrew-Synthetix

> Another phase we discussed is making the EOA -> Smart contract account mapping one-to-many.

Added this to a **Future Work** section

Reply:
No worries, I thought we discussed including some UI / UX spec in the KIP but yea maybe not the right place.